### PR TITLE
Ignore point padding in COPC/laz files

### PR DESCRIPTION
**Pull request by jshrake**
This PR addresses both #39 and #57:

- Fixes the `Header::offset_to_end_of_points` calculation to properly account for the case that the legacy field may be 0.
- Adds test coverage for a COPC file down sampled with PDAL and sourced from https://github.com/PDAL/data/blob/master/autzen/autzen-classified.copc.laz. These tests initially fail, reproducing the error reported in #39.
- Ignores any potential point padding between the end of the point data and the start of the first elvr when the point format is compressed. It's not clear how this check could be supported if we're not able to calculate the uncompressed byte offset of the start of the first elvr.

### CODE REVIEW

File: tests/autzen.rs [hunk 1]
```diff
@@ -90,6 +103,13 @@ fn test_seek_0_works_on_laz() {
     test_seek_0_works_on("tests/data/autzen.laz");
 }
 
+// TODO: This test hangs in LasZipDecompressor::seek
```
Comment (gadomski) [inline tests/autzen.rs:106]: Do we have a guess why this is?

**Review by gadomski — COMMENTED** [review]
Thanks for the contribution! I have one question regarding the test TODOs. And, to confirm, this PR will close #39, correct?

File: tests/autzen.rs (hunk 1 above)
Comment (jshrake) [reply tests/autzen.rs:106]: I didn't debug it any further, but I found a similar issue reported https://github.com/laz-rs/laz-rs/issues/46.

**Comment by jshrake** [comment]
> Thanks for the contribution! I have one question regarding the test TODOs. And, to confirm, this PR will close #39, correct?

Correct, this will close #39.

**Review by gadomski — APPROVED** [review]

**Comment by gadomski** [comment]
Awesome, thanks so much for the contribution! Released: https://crates.io/crates/las/0.8.6

**Pull request merged**